services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/AttachmentController.kt

```diff
@@ -0,0 +1,234 @@
+            ]
+    )
+    fun getAttachment(@PathVariable id: UUID): AttachmentMetadata {
+        val attachmentMetadata = attachmentService.getMetadata(id)
+        val hankeId = hankeService.getHankeId(attachmentMetadata.hankeTunnus)
+
+        checkAttachmentPermission(hankeId, currentUserId(), VIEW)
+
+        return attachmentMetadata
+    }
+
+    @GetMapping("liite/{id}/content")
+    @Operation(summary = "Download attachment file.", description = "Returns the uploaded file.")
+    @ApiResponses(
+        value =
+            [
+                ApiResponse(
+                    description = "Download content of the attachment file.",
+                    responseCode = "200"
+                ),
+                ApiResponse(
+                    description = "Attachment not found.",
+                    responseCode = "404",
+                    content =
+                        [
+                            Content(
+                                schema = Schema(implementation = AttachmentNotFoundException::class)
+                            )
+                        ]
+                ),
+            ]
+    )
+    fun getAttachmentContent(
+        @PathVariable id: UUID,
+        response: HttpServletResponse
+    ): ResponseEntity<ByteArray> {
+        val attachmentMetadata = attachmentService.getMetadata(id)
+        val hankeId = hankeService.getHankeId(attachmentMetadata.hankeTunnus)
+
+        checkAttachmentPermission(hankeId, currentUserId(), VIEW)
+
+        val metadata = attachmentService.getMetadata(id)
+        val mimeType = URLConnection.guessContentTypeFromName(metadata.fileName)
+
+        val responseHeaders =
+            HttpHeaders().apply {
+                contentType = MediaType.parseMediaType(mimeType)
+                add(CONTENT_DISPOSITION, "attachment; filename=${metadata.fileName}")
+            }
+
+        val file = attachmentService.getContent(id)
+        return ResponseEntity.ok().headers(responseHeaders).body(file)
+    }
+
+    @DeleteMapping("liite/{id}")
+    @ApiResponses(
+        value =
+            [
+                ApiResponse(description = "Removes the attachment.", responseCode = "200"),
+                ApiResponse(
+                    description = "Attachment was not found by id",
+                    responseCode = "404",
+                    content =
+                        [
+                            Content(
+                                schema = Schema(implementation = AttachmentNotFoundException::class)
+                            )
+                        ]
+                ),
+            ]
+    )
+    fun removeAttachment(@PathVariable id: UUID) {
+        val info = attachmentService.getMetadata(id)
+        val hankeId = hankeService.getHankeId(info.hankeTunnus)
+
+        checkAttachmentPermission(hankeId, currentUserId(), EDIT)
+
+        return attachmentService.removeAttachment(id)
+    }
+
+    @GetMapping("/hanke/{hankeTunnus}/liite")
```

Comment:
The path is `/hankkeet/` in HankeController. Paths here should follow the same convention.

Reply:
Fixed.